codeql/windows-drivers/queries/Windows/wdk/wdm/InconsistentDispatchAnnotations.ql

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @name Inconsistent annotation of WDM dispatch routine.
+ * @description One or more WDM dispatch routines is incorrectly annotated.
+ * @platform Desktop
+ * @feature.area Multiple
+ * @repro.text One or more WDM dispatch routines is incorrectly annotated.
+ * @kind problem
+ * @id cpp/windows/wdk/inconsistent-dispatch-annotation
+ * @problem.severity warning
+ * @query-version 0.9
+ */
+
+import WdmDrivers
+
+from DispatchTypeDefinition dmi, WdmDispatchRoutine wdr, FunctionDeclarationEntry fde
+where fde = wdr.getADeclarationEntry() and
+dmi.getDeclarationEntry() = fde and
+not (wdr.matchesAnnotation(dmi))
+select fde, "Function declaration of " + fde.getFunction().getName() + " is annotated as a different dispatch type than it is assigned as in the driver's DRIVER_INITIALIZE function ($@).", wdr.getDriverEntry(), wdr.getDriverEntry().getName()
```

Comment:
This would require some extra refactoring/library functions to properly print what the driver was assigned to so I'd prefer not to block on that for the moment - for now I've expanded the message to be more clear and indicate what the mistaken annotation is and what to review.